mlir/test/Dialect/Affine/affine-loop-interchange.mlir

```diff
@@ -0,0 +1,264 @@
+// RUN: mlir-opt %s -affine-loop-interchange | FileCheck %s
+
+func @interchange_for_spatial_locality(%A: memref<2048x2048xf64>) {
+  affine.for %i = 0 to 2048 {
+    affine.for %j = 0 to 2048 {
+      %v = affine.load %A[%j, %i] : memref<2048x2048xf64>
+      affine.store %v, %A[%j, %i] : memref<2048x2048xf64>
+    }
+  }
+  return
+}
+
+// Interchanged for spatial locality.
+// CHECK:       affine.load %arg0[%arg1, %arg2] : memref<2048x2048xf64>
+// CHECK-NEXT:  affine.store %{{.*}}, %arg0[%arg1, %arg2] : memref<2048x2048xf64>
+
+// -----
+
+func @interchange_for_spatial_temporal(%A: memref<2048xf64>) {
+  affine.for %i = 0 to 2048 {
+    affine.for %j = 0 to 2048 {
+      %v = affine.load %A[%j] : memref<2048xf64>
+      affine.load %A[%j] : memref<2048xf64>
+      affine.load %A[%i] : memref<2048xf64>
+    }
+  }
+  return
+}
+
+// More reuse with %j, %i order.
+// CHECK:       affine.load %arg0[%arg1] : memref<2048xf64>
+// CHECK-NEXT:  affine.load %arg0[%arg1] : memref<2048xf64>
+// CHECK-NEXT:  affine.load %arg0[%arg2] : memref<2048xf64>
+
+// -----
+
+// CHECK-LABEL: func @matmul_ijk
+func @matmul_ijk(%A: memref<2048x2048xf64>, %B: memref<2048x2048xf64>, %C: memref<2048x2048xf64>) {
+  affine.for %i = 0 to 2048 {
+    affine.for %j = 0 to 2048 {
+      affine.for %k = 0 to 2048 {
+        %a = affine.load %A[%i, %k] : memref<2048x2048xf64>
+        %b = affine.load %B[%k, %j] : memref<2048x2048xf64>
+        %ci = affine.load %C[%i, %j] : memref<2048x2048xf64>
+        %p = mulf %a, %b : f64
+        %co = addf %ci, %p : f64
+        affine.store %co, %C[%i, %j] : memref<2048x2048xf64>
+      }
+    }
+  }
+  return
+}
+
+// Test whether the ikj permutation has been found.
+
+// CHECK:      affine.load %arg0[%arg3, %arg4] : memref<2048x2048xf64>
+// CHECK-NEXT: affine.load %arg1[%arg4, %arg5] : memref<2048x2048xf64>
+// CHECK-NEXT: affine.load %arg2[%arg3, %arg5] : memref<2048x2048xf64>
+// CHECK-NEXT: mulf %0, %1 : f64
+// CHECK-NEXT: addf %2, %3 : f64
```

Comment:
Do not use numbers in test cases - they can't be maintained easily. Either capture values and compare or use %{{.*}} if you don't want to check for them.

Reply:
Fixed. Thanks.